### ISSUE TITLE
QDialog instantiation generate Vue warning (QV2)

### ISSUE THREAD

**Issue by Sacrekin** [quasarframework/quasar#10391]
» Dev mode.......... pwa
 » Pkg quasar........ v2.0.3
 » Pkg @quasar/app... v3.0.4
 » Transpiled JS..... yes (Babel)

I have a QDialog defined as such:
<template>
  <q-dialog
    v-model="showInfo"
    transition-show="scale"
    transition-hide="scale"
    position="right"
  >
    <q-card style="width: 300px">
      <q-card-section class="row items-center no-wrap">
        <div>
          <q-select>
...

At run time, Vue 3 issues this warning:

 [Vue warn]: Invalid prop: type check failed for prop "onUpdate:modelValue". Expected Function, got Array  
  at <QDialog modelValue=true onUpdate:modelValue= (2) [ƒ, ƒ] transition-show="scale"  ... > 
  at <HoraryTip key=0 modelValue=true onUpdate:modelValue=fn > 
  at <QPage padding="" class="q-pa-md" > 
  at <HoraryPage onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< Proxy {setAnalysis: ƒ, getHoraryPoints: ƒ, pltIcon: ƒ, chartLabel: ƒ, chartIt: ƒ, …} > > 
  at <KeepAlive> 
  at <RouterView> 
  at <QPageContainer> 
  at <QLayout view="lHh Lpr lFf" class="" > 
  at <MwLayout onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< Proxy {checkUpdate: ƒ, help: ƒ, installApp: ƒ, print: ƒ, …} > > 
  at <RouterView> 
  at <App>

**Comment by mila76** [comment]
Please show all the code of the vue component...
What is showInfo ?

[here a little demo](https://codesandbox.io/s/cool-wind-1s84m?file=/src/pages/Index.vue)

**Comment by krisanalfa** [comment]
@mila76 I created a sandbox to reproduce this issue. Checkout the sandbox [here](https://codesandbox.io/s/xenodochial-silence-8d4my?file=/src/pages/Index.vue).

## Error
```
Vue warn]: Invalid prop: type check failed for prop "onUpdate:modelValue". Expected Function, got Array  
  at <QDialog modelValue=false onUpdate:modelValue= (2) [ƒ, ƒ] >
```

## Workaround

Component
```vue
<template>
  <q-dialog
    v-model="model"
    transition-show="scale"
    transition-hide="scale"
    position="right"
  >
    <q-card>
      <q-card-section>
        <div class="text-h6">Alert</div>
      </q-card-section>

      <q-card-section class="q-pt-none">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Rerum
        repellendus sit voluptate voluptas eveniet porro. Rerum blanditiis
        perferendis totam, ea at omnis vel numquam exercitationem aut, natus
        minima, porro labore.
      </q-card-section>

      <q-card-actions align="right">
        <q-btn flat label="OK" color="primary" v-close-popup />
      </q-card-actions>
    </q-card>
  </q-dialog>
</template>

<script>
import { useVModel } from '@vueuse/core'

export default {
  name: 'MyDialog',
  props: {
    value: Boolean
  },
  emits: ['update:value'],
  setup(props, { emit }) {
    const model = useVModel(props, 'value', emit)

    return { model }
  }
}
</script>
```

Usage:
```vue
<template>
  <q-page class="flex flex-center">
    <!-- This will throw warning -->
    <my-dialog v-model="showInfo" />

    <!-- This one won't -->
    <!-- <my-dialog :value="showInfo" /> -->
  </q-page>
</template>

<script>
import { ref } from 'vue'
import MyDialog from '../components/MyDialog.vue'

export default {
  name: 'PageIndex',
  components: { MyDialog },
  setup() {
    return {
      showInfo: ref(true)
    }
  }
}
</script>
```

## Recap

If you make a separate QDialog component like I did in the sandbox, just make sure you use `:value` instead of `v-model`

**Comment by rstoenescu** [comment]
Not sure what `useVModel()` from '@vueuse/core' is actually doing, but the matter of fact is that your “model” is wrongly defined as a result. This is in no way a bug in Quasar.
Rather than using useVModel() you could create a model with getter and setter which passes through the modelValue:

```js
const model = computed({
  get: () => props.modelValue,
  set: val => {
    emit(‘update:modelValue’, val)
  }
})

return { model }
```

Also, it should be:
```js
props: {
  modelValue: Boolean
},
emits: ['update:modelValue'],
```

**Comment by rorgonas** [comment]
This is still a bug, perfectly defined at the end by @rstoenescu in this article:
https://githubmemory.com/repo/quasarframework/quasar/issues/11057

"Hi All,

Very good catch! Fix will be available in Quasar v2.2.4.

v-model is syntactic sugar for model-value="myModel" @update:model-value="val => { myModel = val }" Now when using v-model AND @update:model-value what Vue compiler does is it's setting onUpdate:modelValue to an Array with multiple handlers. But I've defined the type as Function instead of "Function OR Array"."

**Comment by rstoenescu** [comment]
@rorgonas Do you mean that you are you still encountering this in Quasar v2.3.2? If so, can you post a codepen reproducing the issue pls?
